siobrultech_protocols/gem/fields.py

```diff
@@ -4,9 +4,20 @@
 """
 from abc import ABC, abstractmethod
 from datetime import datetime
-from typing import Any, Callable, List
+from enum import Enum
+from typing import Any, List
 
-OrderFn = Callable[[bytes], int]
+
+class ByteOrder(Enum):
+    # Big-endian (the name comes from the GEM packet format spec)
+    HiToLo = 1
+    # Little endian (the name comes from the GEM packet format spec)
+    LoToHi = 2
+
+
+class Sign(Enum):
```

Comment:
We should do [`@unique`](https://docs.python.org/3/library/enum.html#enum.unique) on this `enum` as well